test/TestProjects/FirstTest-Cadl/Generated/Models/Thing.cs

```diff
@@ -16,24 +16,27 @@ public partial class Thing
         /// <summary> Initializes a new instance of Thing. </summary>
         /// <param name="name"></param>
         /// <param name="requiredUnion"></param>
-        /// <param name="requiredLiteral"></param>
-        /// <exception cref="ArgumentNullException"> <paramref name="name"/>, <paramref name="requiredUnion"/> or <paramref name="requiredLiteral"/> is null. </exception>
-        public Thing(string name, string requiredUnion, string requiredLiteral)
+        /// <exception cref="ArgumentNullException"> <paramref name="name"/> or <paramref name="requiredUnion"/> is null. </exception>
+        public Thing(string name, string requiredUnion)
         {
             Argument.AssertNotNull(name, nameof(name));
             Argument.AssertNotNull(requiredUnion, nameof(requiredUnion));
-            Argument.AssertNotNull(requiredLiteral, nameof(requiredLiteral));
 
             Name = name;
             RequiredUnion = requiredUnion;
-            RequiredLiteral = requiredLiteral;
         }
 
         /// <summary> Gets or sets the name. </summary>
         public string Name { get; set; }
         /// <summary> Gets or sets the required union. </summary>
         public string RequiredUnion { get; set; }
-        /// <summary> Gets or sets the required literal. </summary>
-        public string RequiredLiteral { get; set; }
+        /// <summary> Gets the required literal string. </summary>
+        internal string RequiredLiteralString { get; } = "accept";
```

Comment:
well in mgmt plane or hlc, this is implemented inside the constructor using "propertyInitializer" like this: https://github.com/Azure/autorest.csharp/blob/80418efe80837f8748227f19ccf32871c12e7391/src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs#L226

but in our implementation, we are not using this way by adding its initialization to the initializer of the corresponding property while building the ctor.

Do we need to align?

Reply:
This is another option. Previously I found initializers which are bound to constructor parameters, which is not feasible.

Reply:
We have the following codes in `ModelWriter`, that's why I choose to use `DefaultValue`:
https://github.com/Azure/autorest.csharp/blob/80418efe80837f8748227f19ccf32871c12e7391/src/AutoRest.CSharp/Common/Generation/Writers/ModelWriter.cs#L211-L219

Reply:
Because for literal types, the values are constants. So maybe that's enough.